__tests__/ship.test.js

```diff
@@ -0,0 +1,70 @@
+const Ship = require("../src/ship");
+
+describe("Ship", () => {
+  describe("with ports and an itinerary", () => {
+    let ship;
+    let newcastle;
+    let amsterdam;
+    let itinerary;
+
+    beforeEach(() => {
+      newcastle = {
+        addShip: jest.fn(),
+        removeShip: jest.fn(),
+        name: "Newcastle",
+        ships: [],
+      };
+
+      amsterdam = {
+        addShip: jest.fn(),
+        removeShip: jest.fn(),
+        name: "Amsterdam",
+        ships: [],
+      };
+
+      itinerary = {
+        ports: [newcastle, amsterdam],
+      };
+
+      //itinerary = new Itinerary ([newcastle, amsterdam]);
```

Comment:
Rather than creating the itinerary object from scratch like this, we should make use of the constructor on the `itinerary` class as you have in the commented out block here. This would work just fine if you imported the itinerary as you do with `Ship` at the start of the file, delete the block on lines 25-27 and uncomment this out